### PR TITLE
Fix Sentry client setup

### LINKED ISSUE #5296: Sentry release is not set in the Nuxt client
## Problem

The Nuxt Sentry setup now publishes the release, but the client does not send the release name.

## Description

Correctly set release name in the client setup. The `appConfig` cannot be used to set this because this value is undefined when setup the way Sentry does it.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #5296 by @obulat

## Description

This PR changes the way the `release` value is set for the Sentry module on the client. It uses a `sentryRelease` value in the `nuxtConfig` set to the `process.env.SEMANTIC_VERSION`. Using a different name for the variable (i.e., not the corresponding `NUXT_PUBLIC_SENTRY_RELEASE` one) makes sure that the value is not changed during the runtime.

This PR also adds a `platform` tag to easily filter server/client events in Sentry UI.

## Testing Instructions

To test that the release is baked in during the build time, you can use different values for `SEMANTIC_VERSION` env variable during build/runtime. To do that:
1. Build the app using `ov env SEMANTIC_VERSION=build just frontend/run build`
2. Start the app with a different value: `ov env SEMANTIC_VERSION=start just frontend/run build`
3. Go to `http://localhost:8443/preferences`, scroll down to see the "Capture Sentry Exception" button and click it. In the devtools, check the payload of the Sentry POST request. It should have `release: "build"` in it.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -108,7 +108,7 @@
     "npm-run-all2": "^7.0.0",
     "nuxt": "^3.14.1592",
     "rimraf": "^6.0.1",
-    "storybook": "^8.3.6",
+    "storybook": "8.3.6",
```
Comment (obulat) [inline frontend/package.json:111]: The CI was failing because it was using the latest Storybook version

File: frontend/server/api/sources.ts [hunk 2]
```diff
@@ -19,14 +19,16 @@ const getSources = defineCachedFunction(
...
-    consola.info(`Fetching sources for ${mediaType} media`)
+    consola.info(`Fetching ${mediaType} sources.`)
 
     return await $fetch<MediaProvider[]>(
       `${apiUrl}v1/${mediaSlug(mediaType)}/stats/`,
       {
         headers: {
           ...getProxyRequestHeaders(event),
           ...userAgentHeader,
+          "Content-Type": "application/json",
```
Comment (obulat) [inline frontend/server/api/sources.ts:30]: Added the headers to handle the fluke when sometimes the API returns HTML response.

File: frontend/package.json (hunk 1 above)
Comment (dhruvkb) [reply frontend/package.json:111]: What a weird error to have!

File: frontend/server/api/sources.ts [hunk 3]
```diff
@@ -19,14 +19,16 @@ const getSources = defineCachedFunction(
   async (mediaType: SupportedMediaType, event: H3Event) => {
     const apiUrl = useRuntimeConfig(event).public.apiUrl
 
-    consola.info(`Fetching sources for ${mediaType} media`)
+    consola.info(`Fetching ${mediaType} sources.`)
```
Comment (dhruvkb) [inline frontend/server/api/sources.ts:22]: Thanks for making the logs more grammatically correct 😁.

**Review by dhruvkb — APPROVED** [review]
LGTM, appreciate the additional button to test this from the preferences page.

**Pull request merged**